PyInstaller/compat.py

```diff
@@ -39,7 +39,7 @@
 
 is_win = sys.platform.startswith('win')
 is_win_10 = is_win and (platform.win32_ver()[0] == '10')
-is_cygwin = sys.platform == 'cygwin'
+is_cygwin = sys.platform in ('cygwin', 'msys')
```

Comment:
Is this change related to the PR or it just slipped in? 

If `'msys'` is valid `sys.platform` value for Cygwin, then this change affects [this block](https://github.com/pyinstaller/pyinstaller/blob/fee253f15946b31aab629e12ed135a81f730648f/PyInstaller/compat.py#L227-L230) and probably needs corresponding adjustment [here](https://github.com/pyinstaller/pyinstaller/blob/fee253f15946b31aab629e12ed135a81f730648f/bootloader/wscript#L79).

Reply:
Ahh, that's an accident. I seem to need it to just to get PyInstaller to load on MSYS2 but, since there have been no other MSYS2 users reporting this, I'm not too sure if it should be there.

Reply:
Could be related to MSYS2 version, perhaps? 

On my test system, installing and running python in "MSYS2 MSYS" shell (as opposed to "MinGW 32-bit" or "MinGW 64-bit" one) ends up reporting 'cygwin':

```
$ python
Python 3.9.5 (default, May 15 2021, 20:34:13)
[GCC 10.2.0] on cygwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import sys
>>> import platform
>>> sys.platform
'cygwin'
>>> platform.platform()
'MSYS_NT-10.0-19043-3.2.0-340.x86_64-x86_64-64bit-WindowsPE'
```

Reply:
It's quite stubbornly `msys` for me no matter which shell I use:
```
Brénainn@Ostrich MSYS ~
$ python
Python 3.8.6 (default, Oct 23 2020, 14:59:35)
[GCC 9.3.0] on msys
Type "help", "copyright", "credits" or "license" for more information.
>>> import platform
>>> platform.platform()
'MSYS_NT-6.2-9200-3.1.7-340.x86_64-x86_64-64bit-WindowsPE'
>>> import sys
>>> sys.platform
'msys'
```

Reply:
Just updated Python (`pacman -S python`) and now it's switched to cywin.